CUDA device linking is not enabled on windows when building static libraries

### 描述问题

https://gitlab.kitware.com/cmake/cmake/-/issues/17520

cuda在linux下编译静态库时无需进行device linking，但在windows上无论静态库还是动态库都必须进行device linking。xmake这里犯了和cmake同样的错误，windows上编译静态库时没开device linking，导致报错

### 期待的结果

windows上默认打开device linking；

希望能通过设置`cuda.build.devlink`，手动强制启用device linking

### 错误信息

如果是xmake相关编译问题，请加上`-vD`参数运行，并给出完整编译输出信息以及执行的完整命令，例如：

```console
$ xmake -vD
error LNK2019: unresolved external symbol __cudaRegisterLinkedBinary_...
```

### 相关环境

请提供编译和运行环境信息，下面是一些必须填写的基础信息，便于我们针对性排查问题：

- xmake版本：2.6.2
- xmake运行平台：windows
- xmake目标平台：windows

<!--
如果想进一步探讨xmake相关问题，可以加入我们的技术社区互相交流：

* 社区：[Reddit论坛](https://www.reddit.com/r/xmake/)
* 聊天：[Telegram群组](https://t.me/tbooxorg), [Gitter聊天室](https://gitter.im/xmake-io/xmake)
* QQ群：343118190(技术支持), 662147501
* 微信公众号：tboox-os （可通过此公众号申请加入我们的微信群）

注：作者个人精力有限，还要工作吃饭，如果无法第一时间处理问题，请自行提 pr 解决或者耐心等待，也可以考虑赞助项目获取更好的支持。
-->

--- a/xmake/rules/cuda/devlink/xmake.lua
+++ b/xmake/rules/cuda/devlink/xmake.lua
@@ -41,9 +41,10 @@ rule("cuda.build.devlink")
             return
         end
 
-        -- only for binary/shared
+        -- only for binary/shared on non-windows platforms
+        -- https://github.com/xmake-io/xmake/issues/1976
         local targetkind = target:kind()
-        if targetkind ~= "binary" and targetkind ~= "shared" then
+        if target:values("cuda.build.devlink") != true and not target:is_plat("windows") and targetkind ~= "binary" and targetkind ~= "shared" then
             return
         end
 

--- a/xmake/rules/cuda/devlink/xmake.lua
+++ b/xmake/rules/cuda/devlink/xmake.lua
@@ -37,14 +37,14 @@ rule("cuda.build.devlink")
         import("utils.progress")
 
         -- disable devlink?
-        if target:values("cuda.build.devlink") == false then
+        local devlink = target:values("cuda.build.devlink")
+        if devlink == false then
             return
         end
 
         -- only for binary/shared on non-windows platforms
         -- https://github.com/xmake-io/xmake/issues/1976
-        local targetkind = target:kind()
-        if target:values("cuda.build.devlink") != true and not target:is_plat("windows") and targetkind ~= "binary" and targetkind ~= "shared" then
+        if not (devlink == true or target:is_plat("windows") or target:is_binary() or target:is_shared()) then
             return
         end
 
@@ -55,7 +55,7 @@ rule("cuda.build.devlink")
         local culdflags = {"-dlink"}
 
         -- add shared flag
-        if targetkind == "shared" then
+        if target:is_shared() then
             table.insert(culdflags, "-shared")
         end
 

